pymodbus/client/base.py

```diff
@@ -122,10 +122,10 @@ def __init__(  # pylint: disable=too-many-arguments
         self.transaction = DictTransactionManager(
             self, retries=retries, retry_on_empty=retry_on_empty, **kwargs
         )
-        self.reconnect_delay_current = self.params.reconnect_delay
+        self.reconnect_delay_current = cast(float, self.params.reconnect_delay)
```

Comment:
both delay_current and reconnect_delay are (or should be float) so why the cast.

cast is an ugly beast, that preferable never should be used, it typically hides a fundamental problem, likely maybe in this case.

Reply:
`self.params.reconnect_delay` defaults to `None`. Maybe it shouldn't.

Reply:
Yes it should

Reply:
Ok, but if `self.params.reconnect_delay` can be `None` it should not be assigned to `self.reconnect_delay_current` which is declared to be a `float` in 

```
self.reconnect_delay_current = self.params.reconnect_delay
```